browser/net/search_ads_status_header_network_delegate_helper.cc

```diff
@@ -3,40 +3,54 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-#include "brave/browser/net/brave_ads_status_header_network_delegate_helper.h"
+#include "brave/browser/net/search_ads_status_header_network_delegate_helper.h"
 
+#include "brave/components/brave_ads/core/public/prefs/pref_names.h"
 #include "brave/components/brave_rewards/common/pref_names.h"
 #include "brave/components/brave_search/common/brave_search_utils.h"
 #include "chrome/browser/profiles/profile.h"
 #include "components/prefs/pref_service.h"
 #include "content/public/browser/browser_context.h"
 #include "net/base/net_errors.h"
 
+namespace {
+
+bool SearchAdsEnabledForProfile(Profile* profile) {
+  if (!profile || profile->IsOffTheRecord()) {
+    return true;
+  }
+  const auto* prefs = profile->GetPrefs();
+  if (!prefs->GetBoolean(brave_rewards::prefs::kEnabled)) {
+    return true;
+  }
+  return prefs->GetBoolean(brave_ads::prefs::kOptedInToSearchResultAds);
+}
```

Comment:
@aseren over to you :-)

Reply:
The logic looks good. We enable search ads if:
- Profile is private 
- Rewards is disabled
- User opted-in to search ads when Rewards is enabled